### DISCUSSION TITLE
Trying to get immich working on FreeBSD 13.2 .. getting an authorization issue

### DISCUSSION THREAD

**Discussion by ps23Rick** [immich-app/immich#5070] [General]
Hi all ..  So I'm trying to get immich up and have it mostly .. but the server piece is not happy for some reason as I'm not well versed on it's architecture to know where to start diagnosing... Any pointing in the right direction would be great!  I believe that when I got this error, I was uploading a few photos to see if it was working... the photos never showed up and are sitting in the "uploads" folder..

Below is what I found in my /var/log/debug.log for FreeBSD 👍 

```
Nov 15 07:38:08 homegallery immich_server[91370]: UnauthorizedException: Authentication required
Nov 15 07:38:08 homegallery immich_server[91370]:     at AuthService.validate (/usr/local/www/immich/server/dist/domain/auth/auth.service.js:112:15)
Nov 15 07:38:08 homegallery immich_server[91370]:     at CommunicationRepository.handleConnection (/usr/local/www/immich/server/dist/infra/repositories/communication.repository.js:30:49)
Nov 15 07:38:08 homegallery immich_server[91370]:     at Object.next (/usr/local/www/immich/server/node_modules/@nestjs/websockets/web-sockets-controller.js:77:47)
Nov 15 07:38:08 homegallery immich_server[91370]:     at ConsumerObserver.next (/usr/local/www/immich/server/node_modules/rxjs/dist/cjs/internal/Subscriber.js:113:33)
Nov 15 07:38:08 homegallery immich_server[91370]:     at Subscriber._next (/usr/local/www/immich/server/node_modules/rxjs/dist/cjs/internal/Subscriber.js:80:26)
Nov 15 07:38:08 homegallery immich_server[91370]:     at Subscriber.next (/usr/local/www/immich/server/node_modules/rxjs/dist/cjs/internal/Subscriber.js:51:18)
Nov 15 07:38:08 homegallery immich_server[91370]:     at /usr/local/www/immich/server/node_modules/rxjs/dist/cjs/internal/operators/distinctUntilChanged.js:18:28
Nov 15 07:38:08 homegallery immich_server[91370]:     at OperatorSubscriber._this._next (/usr/local/www/immich/server/node_modules/rxjs/dist/cjs/internal/operators/OperatorSubscriber.js:33:21)
Nov 15 07:38:08 homegallery immich_server[91370]:     at Subscriber.next (/usr/local/www/immich/server/node_modules/rxjs/dist/cjs/internal/Subscriber.js:51:18)
Nov 15 07:38:08 homegallery immich_server[91370]:     at /usr/local/www/immich/server/node_modules/rxjs/dist/cjs/internal/Subject.js:69:34
```

it's kicking out on the last line of this method:

```
  async validate(headers: IncomingHttpHeaders, params: Record<string, string>): Promise<AuthUserDto> {
    const shareKey = (headers['x-immich-share-key'] || params.key) as string;
    const userToken = (headers['x-immich-user-token'] ||
      params.userToken ||
      this.getBearerToken(headers) ||
      this.getCookieToken(headers)) as string;
    const apiKey = (headers[IMMICH_API_KEY_HEADER] || params.apiKey) as string;

    if (shareKey) {
      return this.validateSharedLink(shareKey);
    }

    if (userToken) {
      return this.validateUserToken(userToken);
    }

    if (apiKey) {
      return this.validateApiKey(apiKey);
    }

    throw new UnauthorizedException('Authentication required');
  }
```

**Comment by alextran1502** [comment]
Hello, I assume with FreeBSD, you build everything from source, right?

**Reply by ps23Rick** [reply]
Yes.. Someone on the TrueNas forums (link [here](https://www.truenas.com/community/threads/immich-on-truenas-core.113912/#post-791096)) wrote up some steps to walk through and I'm really close...  I'd love to see this working properly even if we have to make a few adjustments here and there for FreeBSD.  If it matters, I'm running it in a FreeBSD Jail.

**Reply by jrasm91** [reply]
The error you linked is an auth error during a websocket connection. There are some places the web tries to connect without being authenticated. Some have been fixed in recent versions.

**Reply by ps23Rick** [reply]
Nevermind.. please ignore!  I found that my micro services was NOT running!  Doh!  sorry!!  All is working just fine now!

**Comment by kenrmayfield** [comment]
@ps23Rick 

Mr. Rick can you provide Your Documentation on the Successful Install of Immich on FreeBSD?

**Did you use this Documention?** >>>>> Immich on TrueNAS Core: 
https://www.truenas.com/community/threads/immich-on-truenas-core.113912/

I took a look at Immich for NetBSD however when Installing  npm install provided Errors in Servers, WEB, Machine-Language and Micro:
Services: https://ipv6.rs/tutorial/NetBSD/Immich/

**Comment by ps23Rick** [comment]
Hi all..

Unfortunately I no longer have a FreeBSD installation at this time so 
I’m not going to be able to assist.  Thank you!
